xapian-applications/omega/postprocess

```diff
@@ -0,0 +1,246 @@
+
+        test_clicks.write('821f03288846297c2cf43c34766a38f7,54\n')
+        test_clicks.write('821f03288846297c2cf43c34766a38f7,54\n')
+        test_clicks.write('098f6bcd4621d373cade4e832627b4f6,35\n')
+        test_clicks.flush()
+
+        test_final = tempfile.NamedTemporaryFile(delete=True)
+
+        generate_combined_log(test_search.name, test_clicks.name, test_final.name)
+
+        test_final.flush()
+
+        # Test entries in final.log are correct.
+        with open(test_final.name, 'r') as final_f:
+            reader = csv.reader(final_f)
+
+            # Skip header row.
+            next(final_f)
+
+            for i, row in enumerate(reader):
+                if i == 0:
+                    assert row == ['821f03288846297c2cf43c34766a38f7',
+                                  'book', '45,54', '0', '45:0,54:2'], "Incorrect entry in final.log"
+                if i == 1:
+                    assert row == ['098f6bcd4621d373cade4e832627b4f6',
+                                  'test', '35,47', '0', '35:1,47:0'], "Incorrect entry in final.log"
+
+        test_query = tempfile.NamedTemporaryFile(delete=True)
+
+        generate_query_file(test_final.name, test_query.name)
+
+        test_query.flush()
+
+        # Test entries in query.txt are correct.
+        with open(test_query.name, 'r') as query_f:
+            reader = csv.reader(query_f)
+
+            for i, row in enumerate(reader):
+                if i == 0:
+                    assert row == ['821f03288846297c2cf43c34766a38f7','book'], "Incorrect entry in query.txt"
+                if i == 1:
+                    assert row == ['098f6bcd4621d373cade4e832627b4f6','test'], "Incorrect entry in query.txt"
+    finally:
+        # Close all files to safely delete them.
+        test_search.close()
+        test_clicks.close()
+        test_final.close()
+        test_query.close()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='''Postprocess click data files. This script generates the
+        final clickstream log file from input search and click log files and
+        creates query file that can be used by the Xapian Letor module for
+        generating its training files.''')
+    parser.add_argument("search_log", type=str, help="Path to the search.log file.")
+    parser.add_argument("clicks_log", type=str, help="Path to the clicks.log file.")
+    parser.add_argument("final_log", type=str, help="Path to save final.log file.")
+    parser.add_argument("query_file", type=str, help="Path to save query.txt file.")
+    parser.add_argument("--test", help="Run tests for this script.", action='store_true')
+    args = parser.parse_args()
+
+    if args.test:
+        test_functions()
+        exit()
+
+    try:
+        generate_combined_log(args.search_log, args.clicks_log, args.final_log)
+        generate_query_file(args.final_log, args.query_file)
+    except IOError as e:
+        print >> sys.stderr, e
```

Comment:
What you had before was fine, but you need `from __future__ import print_function`. See [this helpful document](http://python-future.org/compatible_idioms.html#print) about writing code that's compatible with both versions of python.